server/routes/v1/user.js

```diff
@@ -1,10 +1,11 @@
 import express from 'express';
 import signup from '<controllers>/signup';
 import existingUserCheck from '<middlewares>/existingUserChecker';
-import userValidations from '<middlewares>/userValidations';
+import { validateUserSignUp, validateUserLogin } from '<middlewares>/authValidations';
 
 const router = express.Router();
 
-router.post('/signup', userValidations, existingUserCheck, signup);
+router.post('/signup', validateUserSignUp, existingUserCheck, signup);
```

Comment:
I feel validateUserSignUp and existingUserCheck should just be one middleware except you are planning to reuse it somewhere else.

Reply:
It is a different function and will not obey SRP if that is done.